pkg/nodeagent/controller/lease/add.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lease
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/clock"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// ControllerName is the name of the controller.
+const ControllerName = "lease"
+
+// AddToManager adds the lease controller with the default Options to the manager.
+func (r *Reconciler) AddToManager(mgr manager.Manager) error {
+	if r.Client == nil {
+		r.Client = mgr.GetClient()
+	}
+	if r.RenewIntervalSeconds == 0 {
+		r.RenewIntervalSeconds = 60
```

Comment:
Where are these `60s` coming from? Is this the same interval the `kubelet` uses?

Reply:
It was a best guess, that the lease is updated frequent enough to see in time that something is going wrong with the node agent, but not creating to many requests to the API server.
With 60 seconds and the `/3` calculation I explained in the other comment, node-agent would update the lease every 20 seconds.

kubelet has a default lease duration of 40 seconds ([ref](https://github.com/kubernetes/kubernetes/blob/36930f4cb605450ccbe3126caad365aa5f51267c/pkg/kubelet/apis/config/v1beta1/defaults.go#L132-L134)) and typically renews it every 10 seconds (`0.25* lease duration).

Reply:
Why don't we use the same settings like kubelet then? Or can you present a good reason for using `60s` instead of `40s`?

Reply:
I will change it to the same values as the kubelet